lib/backend/k8s/resources/profile.go

```diff
@@ -66,22 +71,79 @@ func (c *profileClient) Delete(ctx context.Context, key model.Key, revision stri
 	}
 }
 
+func (c *profileClient) getSaKv(sa *kapiv1.ServiceAccount) (*model.KVPair, error) {
+	kvPair, err := c.ServiceAccountToProfile(sa)
+	if err != nil {
+		return nil, err
+	}
+
+	kvPair.Revision = c.joinRevs("", kvPair.Revision)
+	return kvPair, nil
+}
+
+func (c *profileClient) getServiceAccount(ctx context.Context, name, revision string) (*model.KVPair, error) {
+	namespace, serviceAccountName, err := c.ProfileNameToServiceAccount(name)
+	if err != nil {
+		return nil, err
+	}
+
+	serviceAccount, err := c.clientSet.CoreV1().ServiceAccounts(namespace).Get(serviceAccountName, metav1.GetOptions{ResourceVersion: revision})
+	if err != nil {
+		return nil, err
```

Comment:
Anywhere we're querying the k8s API we'll need to wrap in `k8sErrorToCalico` 

like so: https://github.com/colabsaumoh/libcalico-go/blob/4395fc8d5600d87b740ff82e44af1106b5d9814b/lib/backend/k8s/resources/profile.go#L182